### PR TITLE
Alert update

### LINKED ISSUE #4615: Reduce noise in 2xx failure alert
### Background
One of our alerts is set up to catch responses that return a 200 but have a status=failure ( `Prod HTTP Server 2xx (requests where (failed requests * 100.00 / total requests) >= 1)` ). This alert is usually effective, but can be noisy if failures occur outside of business hours, since it's ratio-based.

### Action requested
Update the alerting threshold to be less noisy. This could be done by adding another threshold (keeping the ratio, but adding logic to say "if the total number of failed alerts is less than X, don't fire the alert) or by introducing separate thresholds for business hours/outside business hours.

### Acceptance Criteria
The alert has been adjusted to be less noisy.

### Additional context
See https://prime-cdc.pagerduty.com/incidents/Q1ORTGIRK1DIUR for an example of a noisy page (5 failures occurred from a single bulk uploader, but it was low enough to fire the alert)

### PR DESCRIPTION
**Pull request by alismx**
# DEVOPS PULL REQUEST

## Related Issue

- resolves #4615 

## Changes Proposed

- This PR updates our alert to require a minimum threshold (currently 25) of failing requests and a minimum percentage of requests (currently 1 percent) to fail.
- I've included a modified query that returns the requests that triggered the alert. Hopefully, the comments make its purpose and usage clear.
- The number returned from the alert query now represents the number of requests that matched the criteria to trigger the alert, this way, we immediately know that count. 

**- If 14 requests fail and that's under 1 percent of total requests, the query will return 0 and not alert because of both thresholds.**

**- If 14 requests fail and that's over 1 percent of total requests, the query will return 0 and not alert because of our minimum count threshold.**

**- If 27 requests fail and that's under 1 percent of requests, the query will return 0 and not alert because of our percentage-based threshold.**

**- If 27 requests fail and that's over 1 percent of requests, the query will return 27 and alert.**

## Testing

- I've tested this environment in dev6. Feel free to check the non-prod service group in pagerduty to find the alert. Feel free to use the following to see the alert trigger and requests that triggered it.

```
requests
| where true
| where toint(resultCode) between (200 .. 299) and timestamp >= (datetime(2023-01-10T22:32:46.0000000Z) - 5m)
| sort by timestamp asc
| summarize alert = iff((todouble(sumif(1, success == false)) * 100 / todouble(count()) >= 1 and sumif(1, success == false) > 24), sumif(1, success == false), 0)
| where alert > 0
//
// You can use the following query to see all the requests that triggered this alert.
// These comments are ignored but the timestamp is updated correctly and should match the timestamp used in the above query.
//
// requests
// | where toint(resultCode) between (200 .. 299) and timestamp >= (datetime(2023-01-10T22:32:46.0000000Z) - 5m) and success == false
// | sort by timestamp asc
```

### Infrastructure
- [ ] Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### Cloud
- [ ] Oncall has been notified if this change is going in after-hours
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### Documentation
- [ ] Any changes to the startup configuration have been documented in the README

### CODE REVIEW

File: ops/services/alerts/app_service_metrics/main.tf [hunk 1]
```diff
@@ -87,8 +87,15 @@ requests
 ${local.skip_on_weekends}
 | where toint(resultCode) between (200 .. 299) and timestamp >= ago(5m)
 | sort by timestamp asc
-| summarize alert = iff((todouble(sumif(1, success == false)) * 100 / todouble(count()) >= ${var.http_2xx_failure_rate_threshold}), 1, 0)
-| where alert == 1
+| summarize alert = iff((todouble(sumif(1, success == false)) * 100 / todouble(count()) >= ${var.http_2xx_failure_rate_threshold} and sumif(1, success == false) > ${var.http_2xx_failed_threshold}), sumif(1, success == false), 0)
+| where alert > 0
+//
+// You can use the following query to see all the requests that triggered this alert.
```
Comment (emmastephenson) [inline ops/services/alerts/app_service_metrics/main.tf:93]: Will this comment appear in the PagerDuty link that is generated when the alert fires? (I don't know too much about that integration, so I'm not sure if this is all it takes to update the link or if the comment will just be stored here in the code)

Comment (alismx) [reply ops/services/alerts/app_service_metrics/main.tf:93]: When you click the link in pagerduty into the azure portal, this comment will appear in the query box. It'll appear like I have pasted in the PR description, with the percentage value, threshold, and timestamps all filled in.

Please let me know if I have fully answered your question!

Comment (emmastephenson) [reply ops/services/alerts/app_service_metrics/main.tf:93]: You did! Thank you :)

**Review by emmastephenson — APPROVED** [review]
Looks great!

**Review by rin-skylight — APPROVED** [review]

**Pull request merged**